source/src/enet_encap/encap.c

```diff
@@ -781,8 +781,14 @@ EipInt16 CreateEncapsulationStructure(const EipUint8 *receive_buffer,
   encapsulation_data->options = GetDintFromMessage(&receive_buffer);
   encapsulation_data->current_communication_buffer_position =
     (EipUint8 *) receive_buffer;
-  return (receive_buffer_length - ENCAPSULATION_HEADER_LENGTH
-          - encapsulation_data->data_length);
+
+  const size_t diff = receive_buffer_length - ENCAPSULATION_HEADER_LENGTH
+          - encapsulation_data->data_length;
+
+  /* Sanity check before casting the return value to EipInt16. */
+  OPENER_ASSERT((INT16_MIN <= diff) && (INT16_MAX >= diff));
```

Comment:
This assert is always false in my test case

Reply:
Also, diff as a size_t can never be a negative value

Reply:
Seems I needed to brush up on C integer conversions. It should be fixed in 628cb63.